### ISSUE TITLE
`rm` is buggy when path contains globs

### ISSUE THREAD

**Issue by horasal** [nushell/nushell#10211]
### Describe the bug

Related: https://github.com/nushell/nushell/issues/9310 

`rm` directly passes `current_directory` to `nu_glob` without escaping, so the path itself is also expanded as globs.

For example, run `rm a` under `/tmp/*` will also delete `/tmp/b/a`, `/tmp/whatever/a`:

```bash
❯ ls **/*
╭────┬──────────┬──────┬───────┬────────────────╮
│  # │   name   │ type │ size  │    modified    │
├────┼──────────┼──────┼───────┼────────────────┤
│  0 │ *        │ dir  │ 192 B │ a minute ago   │
│  1 │ */a      │ file │   0 B │ a minute ago   │
│  2 │ */b      │ file │   0 B │ a minute ago   │
│  3 │ */c      │ file │   0 B │ a minute ago   │
│  4 │ */d      │ file │   0 B │ a minute ago   │
│  5 │ [test]   │ dir  │ 192 B │ 30 seconds ago │
│  6 │ [test]/a │ file │   0 B │ 30 seconds ago │
│  7 │ [test]/b │ file │   0 B │ 30 seconds ago │
│  8 │ [test]/c │ file │   0 B │ 30 seconds ago │
│  9 │ [test]/d │ file │   0 B │ 30 seconds ago │
│ 10 │ a        │ file │   0 B │ 4 minutes ago  │
│ 11 │ b        │ file │   0 B │ 4 minutes ago  │
│ 12 │ c        │ file │   0 B │ 4 minutes ago  │
│ 13 │ d        │ file │   0 B │ 4 minutes ago  │
│ 14 │ e        │ file │   0 B │ 4 minutes ago  │
│ 15 │ f        │ file │   0 B │ 4 minutes ago  │
│ 16 │ g        │ file │   0 B │ 4 minutes ago  │
╰────┴──────────┴──────┴───────┴────────────────╯


❯ cd *; rm a; cd ..; ls **/*
╭────┬──────────┬──────┬───────┬───────────────╮
│  # │   name   │ type │ size  │   modified    │
├────┼──────────┼──────┼───────┼───────────────┤
│  0 │ *        │ dir  │ 160 B │ now           │
│  1 │ */b      │ file │   0 B │ a minute ago  │
│  2 │ */c      │ file │   0 B │ a minute ago  │
│  3 │ */d      │ file │   0 B │ a minute ago  │
│  4 │ [test]   │ dir  │ 160 B │ now           │
│  5 │ [test]/b │ file │   0 B │ a minute ago  │
│  6 │ [test]/c │ file │   0 B │ a minute ago  │
│  7 │ [test]/d │ file │   0 B │ a minute ago  │
│  8 │ a        │ file │   0 B │ 4 minutes ago │
│  9 │ b        │ file │   0 B │ 4 minutes ago │
│ 10 │ c        │ file │   0 B │ 4 minutes ago │
│ 11 │ d        │ file │   0 B │ 4 minutes ago │
│ 12 │ e        │ file │   0 B │ 4 minutes ago │
│ 13 │ f        │ file │   0 B │ 4 minutes ago │
│ 14 │ g        │ file │   0 B │ 4 minutes ago │
╰────┴──────────┴──────┴───────┴───────────────╯
```

### How to reproduce

1. `mkdir * a b c d`
2. `touch */a */b */c */d a/a b/a c/a d/a`
3. `cd *; rm a`

### Expected behavior

Only delete `*/a`

### Configuration

| column0            | column1                                                    |
| ------------------ | ---------------------------------------------------------- |
| version            | 0.84.1                                                     |
| branch             | master                                                     |
| commit_hash        | e5145358eb4a13436eccae58c7e07acc9666b661                   |
| build_os           | macos-aarch64                                              |
| build_target       | aarch64-apple-darwin                                       |
| rust_version       | rustc 1.70.0 (90c541806 2023-05-31)                        |
| rust_channel       | 1.70.0-aarch64-apple-darwin                                |
| cargo_version      | cargo 1.70.0 (ec8a8a0ca 2023-04-25)                        |
| build_time         | 2023-09-04 09:41:32 +09:00                                 |
| build_rust_channel | release                                                    |
| allocator          | standard                                                   |
| features           | dataframe, default, extra, sqlite, trash, wasi, which, zip |
| installed_plugins  | query, query json, query web, query xml                    |

**Comment by fdncred** [comment]
We're planning on replacing `rm` with the uutils equivalent, just like we're doing `cp` right now in this PR https://github.com/nushell/nushell/pull/10097. It's fine if someone wants to work on it but that work may be removed once we migrate to `uu_rm`.

**Comment by bobhy** [comment]
I believe this is fixed by #10571.
